src/Symfony/Component/ExpressionLanguage/Node/GetAttrNode.php

```diff
@@ -105,6 +103,24 @@ public function evaluate(array $functions, array $values)
         }
     }
 
+    private function isEvaluationShortCircuited(array $functions, array $values): bool
+    {
+        $node = $this;
+
+        do {
+            if (
+                $node->nodes['attribute'] instanceof ConstantNode
+                && $node->nodes['attribute']->isNullSafe
+                && null === $node->nodes['node']->evaluate($functions, $values)
```

Comment:
this is going to evaluate the same node many times, isn't it? can't we find a way to prevent this?

Reply:
@nicolas-grekas Yes, but I haven't found an obvious way to avoid this so far. I'll continue to try though :smile:

Reply:
@nicolas-grekas I think this'll do the trick.